src/unlock/omni_lock.rs

```diff
@@ -0,0 +1,158 @@
+use std::fmt::Display;
+
+use crate::{types::omni_lock::OmniLockWitnessLock, util::blake160};
+use ckb_types::{
+    bytes::{BufMut, Bytes, BytesMut},
+    packed::WitnessArgs,
+    prelude::*,
+    H160,
+};
+
+use ckb_crypto::secp::Pubkey;
+pub use ckb_types::prelude::Pack;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum IdentityFlags {
+    PubkeyHash = 0,
+}
+
+impl From<IdentityFlags> for u8 {
+    fn from(val: IdentityFlags) -> Self {
+        val as u8
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Identity {
+    pub flags: IdentityFlags,
+    pub blake160: H160,
+}
+impl Identity {
+    pub fn to_smt_key(&self) -> [u8; 32] {
+        let mut ret: [u8; 32] = Default::default();
+        ret[0] = self.flags.into();
+        (&mut ret[1..21]).copy_from_slice(self.blake160.as_ref());
+        ret
+    }
+}
+
+impl From<Identity> for [u8; 21] {
+    fn from(id: Identity) -> Self {
+        let mut res = [0u8; 21];
+        res[0] = id.flags.into();
+        res[1..].copy_from_slice(id.blake160.as_bytes());
+        res
+    }
+}
+
+impl From<Identity> for Vec<u8> {
+    fn from(id: Identity) -> Self {
+        let mut bytes: Vec<u8> = vec![id.flags.into()];
+        bytes.extend(id.blake160.as_bytes());
+        bytes
+    }
+}
+
+impl From<Identity> for ckb_types::bytes::Bytes {
+    fn from(id: Identity) -> Self {
+        let v: Vec<u8> = id.into();
+        v.into()
+    }
+}
+
+impl Display for Identity {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let v: Vec<u8> = self.clone().into();
+        for i in v {
+            write!(f, "{:02x}", i)?;
+        }
+        Ok(())
+    }
+}
+
+/*
+<1 byte Omnilock flags>
+<32 byte RC cell type ID, optional>
+<2 bytes minimum ckb/udt in ACP, optional>
+<8 bytes since for time lock, optional>
+<32 bytes type script hash for supply, optional>
+*/
+
+#[derive(Clone, Serialize, Deserialize)]
```

Comment:
Add more traits:
* `Debug`
* `Hash` (since it can be used as key of `HashMap`)
* `Eq`, `PartialEq` (since it will be compared to each other)